### PR TITLE
Specify Arrow as an external dependency

### LINKED ISSUE #213: Arrow library is not externalized
:beetle:

In 5.5.1, the arrow library is not externalized in the compiled bundle. That means that if someone wants to check whether the dataframe is an instance of an arrow dataframe, they cannot easily do it. the problem is that the constructors are not the same (the connector has their own version of arrow, not the same that someone else might use to check the type).

Because of this issue, the check in https://github.com/vega/vega-loader-arrow/blob/36e1d0d2b0050f6e207327e78367007c4d25e28b/src/arrow.js#L20 does not work.

### PR DESCRIPTION
**Pull request by bmatcuk**
Do not pack Arrow source files into the final bundle, rely on them being provided externally. This should fix #213.

### CODE REVIEW

**Comment by domoritz** [comment]
Eventually, lodash, thrift and so on should also be external dependencies, right?

**Comment by bmatcuk** [comment]
I'm not sure... can you test if this resolves the issues with arrow?

If you replace the require in your observable notebook with the following, it will load this PR:

```
  main.variable(observer("MapdCon")).define("MapdCon", ["require"], async function(require){return(
(await require("https://cdn.jsdelivr.net/gh/omnisci/mapd-connector@jclay-jclay/arrow_external/dist/browser-connector.js"))
)});
```

When I try it locally, it throws an error in vega-embed: `TypeError: Cannot read property 'type' of null` - I'm not sure where that's coming from.

**Comment by domoritz** [comment]
Nice. With your version, I am getting that the returned object is an arrow table so that's awesome. 

Screen Shot 2021-01-19 at 22 12 18

I have to see where `TypeError: Cannot read property 'type' of null` is coming from now.

**Comment by domoritz** [comment]
Ahh, so the issue is that when Arrow is not es2015. 

See this working example: https://observablehq.com/@vega/vega-lite-and-apache-arrow-no-plugin
And if you change the import to be `@apache-arrow/es5-umd`, it breaks.

Not sure what's going wrong exactly. Do you see a solution?

File: webpack.browser.js [hunk 1]
```diff
@@ -32,6 +31,14 @@ module.exports = {
       }
     ]
   },
+  externals: {
+    "apache-arrow": {
+      commonjs: "apache-arrow",
+      commonjs2: "apache-arrow",
+      amd: "apache-arrow",
```
Comment (domoritz) [inline webpack.browser.js:38]:
```suggestion
      commonjs: "@apache-arrow/es2015-umd",
      commonjs2: "@apache-arrow/es2015-umd",
      amd: "@apache-arrow/es2015-umd",
```

**Comment by domoritz** [comment]
@bmatcuk found the core of the issue to be in Vega and we fixed it in https://github.com/vega/vega/pull/3050. From my side, this is good to go.

File: webpack.browser.js (hunk 1 above)
Comment (domoritz) [reply webpack.browser.js:38]: Not needed anymore.

**Pull request merged**